action: get latest tag by github api

Desiki-high opened the pull request:
## Details
Use [Github API](https://api.github.com/repos/Dragonflyoss/nydus/releases/latest) to get the latest tag of nydus, and used in smoke/integration-test.

Enhancement  https://github.com/dragonflyoss/nydus/pull/1462.

## Types of changes

_What types of changes does your PullRequest introduce? Put an `x` in all the boxes that apply:_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Go over all the following points, and put an `x` in all the boxes that apply._

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

Desiki-high commented:
If the `latest` release is stable, I think this patch can work batter. cc@imeoer.

imeoer commented on .github/workflows/smoke.yml:
```diff
@@ -177,8 +177,11 @@ jobs:
 
-        versions=(v0.1.0 ${{env.NYDUS_STABLE_VERSION}} latest)
-        version_exports=(v0_1_0 ${{env.NYDUS_STABLE_VERSION_EXPORT}} latest)
+        export NYDUS_STABLE_VERSION=$(curl https://api.github.com/repos/Dragonflyoss/nydus/releases/latest | jq -r '.tag_name')
```
Maybe can reuse the env defined above.

Desiki-high replied:
It seems good.

imeoer reviewed (commented):
others LGTM, thanks!

Desiki-high committed:
action: get latest tag by Github API

Use https://api.github.com/repos/Dragonflyoss/nydus/releases/latest to get the
latest tag of nydus, and used in smoke/integration-test.

Signed-off-by: Yadong Ding <ding_yadong@foxmail.com>

Desiki-high commented:
In `Github Action`, the environment variable can't share between different steps in the same job, see https://github.com/dragonflyoss/nydus/actions/runs/6766960684/job/18388873567. So we need to export 
`NYDUS_STABLE_VERSIO` twice. cc @imeoer .

imeoer reviewed (approved):
Thanks!
